src/cocoa/toga_cocoa/window.py

```diff
@@ -221,13 +222,50 @@ def set_content(self, widget):
     def set_title(self, title):
         self.native.title = title
 
+    def update_position(self):
+        screen = self.native.screen
+        window = self.native
+        if screen is not None:
+            y = screen.frame.size.height - (
+                window.frame.origin.y + window.frame.size.height
+            )
+            self.interface._position = (
+                screen.frame.origin.x + window.frame.origin.x,
+                screen.frame.origin.y + y
+            )
+
     def set_position(self, position):
-        pass
+        x, y = position
+        # The "principal" screen has index 0 and origin (0, 0).
+        screen = NSScreen.screens[0]
+
+        # If the window isn't visible, it won't be on a screen;
+        # don't try to set the position.
+        if screen is None:
+            return
+
+        frame = screen.frame
+
+        # From the top left corner of the screen to the top left corner
+        # of the window.
+        x = frame.origin.x + x
+        y = frame.origin.y + frame.size.height - y
+        self.native.setFrameTopLeftPoint(NSPoint(x, y))
+
+        # Update the actual position.
+        self.update_position()
+
+        # If we ended up off screen, or not where requested,
+        # fall back to the "main" screen.
+        if screen is None or not self.interface._position == position:
+            frame = NSScreen.mainScreen.frame
+            self.set_position((frame.origin.x + 100, frame.origin.y + 100))
```

Comment:
Why do you choose the (100, 100) relative to "main screen" as a fallback instead of just accepting the best effort of macOS?

For example, when calling `setPosition((0,0))`, macOS would position the window in the top-left corner and possible move it down by 25 pixels to accommodate the menu bar. This seems like perfectly reasonable behaviour to me and more in line with expectations than moving the window to (100, 100).

Reply:
I was worried the window might disappear off screen, which was silly, AppKit takes care of that.